capz/run-capz-e2e.sh

```diff
@@ -181,6 +182,254 @@ run_e2e_test() {
             --v=5 "${ADDITIONAL_E2E_ARGS[@]}"
         set +x
         log "e2e tests complete"
+
+        if [[ "$INCLUDE_NPM_TESTS" == "true" ]]; then
```

Comment:
Our goal is eventually to test Azure CNI on `az capi`. Support for this is in progress.

This script keeps Calico CNI, although it disables its NetworkPolicy functionality and seems to break inter-node connectivity. This can be restored with `helm install calico ...`

Reply:
to clarify, the script uses Calico CNI + Azure NetPol (NPM)